packages/@aws-cdk/core/lib/arn.ts

```diff
@@ -43,6 +43,14 @@ export interface ArnComponents {
    */
   readonly resource: string;
 
+  /**
+   * Whether the 'resource' part is prefixed with a '/',
+   * such as in an ARN in the format 'arn:partition:service:region:account:/resource/resourceName'.
+   *
+   * @default - false (the 'resource' part does not have a prefix)
+   */
+  readonly resourcePrefixed?: boolean;
```

Comment:
The name could be more descriptive. First time I read this my head went "so what's the prefix?" Or you could also read it it as "*something* is prefixed *with the resource*".

How about (just spitballing here):

- `slashResource`
- `absResource` 
- `resourcePathAbsolute` (bit long maybe)
- `arnFormat3` ?

(Maybe we should just be introducing the concept of ARN format versions...??)

- `resourcePrefix (either '/' or '')`

Reply:
Not sure. Maybe ask some other people for inspiration?

Reply:
I like the ARN format version suggestion.

What version should this be? Is it 2, or 3?

* `arn:partition:service:region:account-id:resource` is version 1, I guess.
* `arn:partition:service:region:account-id:resource:resourceName` is also 1?
* `arn:partition:service:region:account-id:resource/resourceName` is this 1, or 2?
* `arn:partition:service:region:account-id:/resource/resourceName` this is "version nr of the previous one + 1", I guess?

Reply:
Well if we have this lever to pull, we could make it 1, 2, 3, 4. Give ourselves all the information we need...?

Reply:
Done.